examples/custom_descriptor_example.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import torch
+from amptorch.ase_utils import AMPtorch
+from amptorch.descriptor.Gaussian import GaussianDescriptorSet
+from amptorch.trainer import AtomsTrainer
+from ase import Atoms
+from ase.calculators.emt import EMT
+
+distances = np.linspace(2, 5, 100)
+images = []
+for dist in distances:
+    image = Atoms(
+        "CuCO",
+        [
+            (-dist * np.sin(0.65), dist * np.cos(0.65), 0),
+            (0, 0, 0),
+            (dist * np.sin(0.65), dist * np.cos(0.65), 0),
+        ],
+    )
+    image.set_cell([10, 10, 10])
+    image.wrap(pbc=True)
+    image.set_calculator(EMT())
+    images.append(image)
+
+
+elements = np.unique([atom.symbol for atom in images[0]])
+cutoff = 6.0
+cosine_cutoff_params = {"cutoff_func": "cosine"}
+
+Gs = {
+    "default": {
+        "G2": {
+            "etas": np.logspace(np.log10(0.05), np.log10(5.0), num=4),
+            "rs_s": [0],
+        },
+        "G4": {"etas": [0.005], "zetas": [1.0, 4.0], "gammas": [1.0, -1.0]},
+        "cutoff": 6.0,
+    },
+}
+
+gds_combo = GaussianDescriptorSet(elements, cutoff, cosine_cutoff_params)
+gds_combo.process_combinatorial_Gs(
+    Gs
+)  # a GDS can be constructed directly from the `Gs` dict
+
+gds_custom = GaussianDescriptorSet(elements, cutoff, cosine_cutoff_params)
+g2_etas = np.logspace(np.log10(0.05), np.log10(5.0), 4)
+g2_rs_s = np.zeros(g2_etas.shape)
+gds_custom.batch_add_descriptors(
+    2, g2_etas, g2_rs_s, []
+)  # or the descriptors can be manually added into the GDS
+g4_etas = [0.005] * 4
+g4_zetas = [1.0, 4.0, 1.0, 4.0]
+g4_gammas = [1.0, 1.0, -1.0, -1.0]
+gds_custom.batch_add_descriptors(
+    4, g4_etas, g4_zetas, g4_gammas
+)  # this opens opportunities for finely-tuned, custom descriptors
+
+print("both GaussianDescriptorSets are equivalent!")
+print("combo hash", gds_combo.descriptor_setup_hash)
+print("custom hash", gds_custom.descriptor_setup_hash)
+print("gds_combo == gds_custom:", gds_combo == gds_custom)
```

Comment:
Instead, can you make this example demonstrate what the custom `GDS` allows you to do rather than comparing it to the old (you already do that in the test script)? For instance construct fingerprints other than the default method.

Reply:
yessir, good idea